### PR TITLE
CORE:

### PR DESCRIPTION
**Pull request by manolama**
- Start a new DefaultQueryPlanner class that will parse out the ExecutionGraph
  and handle push-down operations to the data source. This will replace the
  old planners and the abstract context planner.

### CODE REVIEW

File: core/src/main/java/net/opentsdb/query/AbstractQueryPipelineContext.java [hunk 1]
```diff
@@ -59,7 +52,7 @@
  * 
  * <b>Invariants:</b>
  * <ul>
- * <li>Each {@link ExecutionGraphNode} must have a unique ID within the graph.</li>
+ * <li>Each {@link ExecutionGraphNode} must have a unique ID within the plan.graph().</li>
```
Comment (SeanPMiller) [inline core/src/main/java/net/opentsdb/query/AbstractQueryPipelineContext.java]: Are these invariants checked statically (can't find the code yet), or do they cause run-time errors?

File: core/src/main/java/net/opentsdb/query/execution/serdes/JsonV2QuerySerdes.java [hunk 2]
```diff
@@ -470,4 +333,152 @@ public String version() {
...
+    }
+    json.writeEndObject();
+    json.writeArrayFieldStart("aggregateTags");
+    for (final String tag : id.aggregatedTags()) {
+      json.writeString(tag);
+    }
+    json.writeEndArray();
+    json.writeObjectFieldStart("dps");
+    
+    if (iterator.getType() == NumericType.TYPE) {
```
Comment (SeanPMiller) [inline core/src/main/java/net/opentsdb/query/execution/serdes/JsonV2QuerySerdes.java]: Hmm... moved the branching on the numeric type to here. Looks good.

File: core/src/test/java/net/opentsdb/query/TestAbstractQueryPipelineContext.java [hunk 3]
```diff
@@ -416,154 +422,146 @@ public void initializeNoSuchUpstreamNode() throws Exception {
...
-    
-    assertNull(n1.config());
-    assertNull(n2.config());
-    assertNull(n3.config());
-    assertSame(config_a, n4.config());
-    assertNull(n5.config());
-    assertNull(s1.config());
-    assertNull(s2.config());
-  }
+//  @Test
```
Comment (SeanPMiller) [inline core/src/test/java/net/opentsdb/query/TestAbstractQueryPipelineContext.java]: Reintroducing this later?

File: core/src/main/java/net/opentsdb/query/execution/serdes/JsonV3QuerySerdes.java [hunk 4]
```diff
@@ -484,4 +259,258 @@ public void deserialize(final SerdesOptions options,
         + "class: " + getClass().getCanonicalName()));
   }
 
+  private void serializeSeries(
```
Comment (SeanPMiller) [inline core/src/main/java/net/opentsdb/query/execution/serdes/JsonV3QuerySerdes.java]: Same changes as V2...

File: core/src/main/java/net/opentsdb/query/AbstractQueryPipelineContext.java (hunk 1 above)
Comment (manolama) [reply core/src/main/java/net/opentsdb/query/AbstractQueryPipelineContext.java]: Yes, throws a runtime error in the new `DefaultQueryPlanner` when it tries to build the graph.

File: core/src/test/java/net/opentsdb/query/TestAbstractQueryPipelineContext.java (hunk 3 above)
Comment (manolama) [reply core/src/test/java/net/opentsdb/query/TestAbstractQueryPipelineContext.java]: yeah, will move it to the planner UTs.

**Comment by manolama** [comment]
Merged in 4e3604bcbc51cbfcc6bf6bfd315975cf2bdf4323

**Pull request closed without merging**